Criar classes singleton para simular as tabelas

RubensAugustoMM opened the pull request.

RubensAugustoMM committed:
Adicionado classe singleton generica e alterado o repositorio mock para trabalhar com ela

RubensAugustoMM committed:
Adicionado classe singleton generica e alterado o repositorio mock para trabalhar com ela

RubensAugustoMM committed:
remove arquivo desnecessário

RubensAugustoMM committed:
Alterado o nome do arquivo 'SingletonTabela' para 'TabelaSingleton'. retirado o modificador de acesso private de todos os metodos set das classes do modelo para permitir a inserção de dados

Ang-Diniz commented on Cod3rsGrowth.Testes/TabelaSingleton.cs:
```diff
@@ -0,0 +1,24 @@
+namespace Cod3rsGrowth.Testes;
+
+public class TabelaSingleton<T> where T : IEntidade
+{
```
A sua classe singleton tem que ser uma instancia da sua propria classe

use esse link como exemplo:
https://refactoring.guru/pt-br/design-patterns/singleton/csharp/example

Ang-Diniz commented on Cod3rsGrowth.Testes/TabelaSingleton.cs:
```diff
@@ -0,0 +1,24 @@
+
+    }
+
+    public static TabelaSingleton<T> Instance
```
Sua classe deve ter como propriedades listas para simular todas as suas tabelas ( uma lista para cada uma das suas tabelas )

RubensAugustoMM committed:
altera classe tabela singleton para uma classe não generica, adiciona o metodo 'RetornaTabela' e altera a classe 'ReposotorioMock'

RubensAugustoMM committed:
Corrige classe TabelaSingleton

RubensAugustoMM committed:
Altera a injeção de dependencia para injetar a interface repositorio

RubensAugustoMM committed:
Altera a injeção de dependencia para injetar a interface repositorio

RubensAugustoMM committed:
corrige erro no moduloInjetor

RubensAugustoMM committed:
corrigido erros de nomenclatura

Ang-Diniz commented on Cod3rsGrowth.Testes/TabelaSingleton.cs:
```diff
@@ -0,0 +1,49 @@
+
+    public List<IEntidade> RetornaTabela<T>() where T : IEntidade
+    {
+        if (typeof(T) == typeof(Convenio))
```
remover esse tanto de if e trocar por um switch case, se quiser alterar a forma de fazer, pode usar lazy, conversar com os meninos ai que eles já estão utilizando

Ang-Diniz commented on Cod3rsGrowth.Testes/TabelaSingleton.cs:
```diff
@@ -0,0 +1,49 @@
+
+    }
+
+    public List<IEntidade> RetornaTabela<T>() where T : IEntidade
```
Vê se consegue fazer sem utilizar esse IEntidade, e remover ele do projeto, pense em alguma forma

RubensAugustoMM committed:
adiciona metodos lazy e retira o bloco de if gigante

RubensAugustoMM committed:
corrige erros

RubensAugustoMM committed:
task-criar-classes-singleton-para-simular-as-tabelas:
Neste commit, foi retirado todas as referências à interface IEntidade.
Com isso, a interface do repositorio IRepositorio e a sua
implementação foi substituida por interfaces e classes de repositório específicas
para cada classe do modelo, o mesmo foi feito para o repositório mock.
Além disso, todas os serviços de repositórios mocks foram adicionados à
injeção de dependencia com o tipo da interface de cada repositório, para
no futuro ser injetado nas dependências do metodo construtor de cada
serviço.
Em questão de estrutura, foram criadas pastas no projeto de
infraestrutura e testes para organizar melhor a solução.
\#time 67h 4.020m

Ang-Diniz commented on Cod3rsGrowth.Testes/Mocks/MockRepositorioConvenio.cs:
```diff
@@ -0,0 +1,39 @@
+    {
+        var objeto = Tabelas._convenios.Value.Find(x => x.Id == entidade.Id);
+
+        Tabelas._convenios.Value.Remove(objeto);
```
não entendi pq está removendo aqui

Ang-Diniz commented on Cod3rsGrowth.Testes/Mocks/MockRepositorioConvenio.cs:
```diff
@@ -0,0 +1,39 @@
+
+    public Empresa ObterPorId(int Id)
+    {
+        return Tabelas._convenios.Value.Find(x => x.Id == Id); 
```
utilizar FirstOrDefault e implementar uma Exception para caso o objeto com esse id não exista

Ang-Diniz commented on Cod3rsGrowth.Testes/Mocks/MockRepositorioEmpresa.cs:
```diff
@@ -0,0 +1,37 @@
+    {
+        var objeto = Tabelas._empresas.Value.Find(x => x.Id == entidade.Id);
+
+        Tabelas._empresas.Value.Remove(objeto);
```
Os mesmos comentários da classe anterior

Ang-Diniz commented on Cod3rsGrowth.Testes/Mocks/MockRepositorioEndereco.cs:
```diff
@@ -0,0 +1,37 @@
+    {
+        var objeto = Tabelas._enderecos.Value.Find(x => x.Id == entidade.Id);
+
+        Tabelas._enderecos.Value.Remove(objeto);
```
Os mesmos comentários da classe anterior

Ang-Diniz commented on Cod3rsGrowth.Testes/Mocks/MockRepositorioEscola.cs:
```diff
@@ -0,0 +1,37 @@
+    {
+        var objeto = Tabelas._escolas.Value.Find(x => x.Id == entidade.Id);
+
+        Tabelas._escolas.Value.Remove(objeto);
```
Os mesmos comentários da classe anterior

Ang-Diniz commented on Cod3rsGrowth.Testes/Mocks/MockRepositorioEstado.cs:
```diff
@@ -0,0 +1,37 @@
+    {
+        var objeto = Tabelas._estados.Value.Find(x => x.Id == entidade.Id);
+
+        Tabelas._estados.Value.Remove(objeto);
```
Os mesmos comentários da classe anterior

RubensAugustoMM committed:
task-criar-classes-singlton-para-simular-as-tabelas: Corrige domínio, corrige nomenclatura nos repositórios e suas interfaces, corrige método ObterPorId em todos os repositórios, corrige método Atualizar em todos os repositórios

RubensAugustoMM committed:
task-criar-classes-singlton-para-simular-as-tabelas: Corrige erro de nomenclatura na classe TabelaSingleton \# time 21h 1260m

Ang-Diniz commented on Cod3rsGrowth.Testes/Mocks/MockRepositorioConvenio.cs:
```diff
@@ -0,0 +1,42 @@
+    TabelaSingleton Tabelas = TabelaSingleton.Instance;
+
+    public void Atualizar(Convenio convenioAtulizado)
+    {
```
aqui não precisa mandar o objeto inteiro, se vai usar só id, mandar apenas o id

Ang-Diniz commented on Cod3rsGrowth.Testes/Mocks/MockRepositorioConvenio.cs:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    public void Deletar(Convenio convenioDeletado)
+    {
```
aqui não precisa mandar o objeto inteiro, vc vai usar apenas o id, pra obter por id com o metodo que já existe, então mandar apenas o id

Ang-Diniz commented on Cod3rsGrowth.Testes/Mocks/MockRepositorioConvenio.cs:
```diff
@@ -0,0 +1,42 @@
+
+    public Convenio ObterPorId(int Id)
+    {
+        return Tabelas._convenios.Value.FirstOrDefault(c => c.Id == Id) ?? throw new NullReferenceException();
```
usar uma Exception com uma mensagem descritiva

Ang-Diniz commented on Cod3rsGrowth.Testes/Mocks/MockRepositorioEmpresa.cs:
```diff
@@ -0,0 +1,47 @@
+        empresaExistente.Porte = empresaAtualizada.Porte;
+        empresaExistente.MatrizFilial = empresaAtualizada.MatrizFilial;
+        empresaExistente.ListaConvenios = empresaAtualizada.ListaConvenios;
+    }
```
os comentários do repositorio anterior servem pra esse tb

Ang-Diniz commented on Cod3rsGrowth.Testes/Mocks/MockRepositorioEndereco.cs:
```diff
@@ -0,0 +1,44 @@
+    {
+        var enderecoExistente = ObterPorId(enderecoAtualizado.Id);
+
+        enderecoExistente.Numero = enderecoAtualizado.Numero;
```
os comentários do repositorio anterior servem pra esse tb

Ang-Diniz commented on Cod3rsGrowth.Testes/Mocks/MockRepositorioEscola.cs:
```diff
@@ -0,0 +1,45 @@
+        escolaExistente.StatusAtividade = escolaAtualizada.StatusAtividade;
+        escolaExistente.Nome = escolaAtualizada.Nome;
+        escolaExistente.CodigoMec = escolaAtualizada.CodigoMec;
+        escolaExistente.Telefone = escolaAtualizada.Telefone;
```
os comentários do repositorio anterior servem pra esse tb

Ang-Diniz commented on Cod3rsGrowth.Testes/Mocks/MockRepositorioEstado.cs:
```diff
@@ -0,0 +1,38 @@
+
+        estadoExistente.Nome = estadoAtualizado.Nome;
+        estadoExistente.Sigla = estadoAtualizado.Sigla;
+        estadoExistente.ListaEnderecos = estadoAtualizado.ListaEnderecos;
```
os comentários do repositorio anterior servem pra esse tb

RubensAugustoMM committed:
task-criar-classes-singlton-para-simular-as-tabelas: Corrige função deletar e ObterPorId \# time 00h 57m
